src/phpDocumentor/Pipeline/Stage/Transform.php

```diff
@@ -87,26 +88,32 @@ public function __invoke(Payload $payload): Payload
         $project = $payload->getBuilder()->getProjectDescriptor();
         $transformations = $templates->getTransformations();
 
-        /** @var PreTransformEvent $preTransformEvent */
-        $preTransformEvent = PreTransformEvent::createInstance($this);
-        $preTransformEvent->setProject($project);
-        $preTransformEvent->setTransformations($transformations);
-        Dispatcher::getInstance()->dispatch(
-            $preTransformEvent,
-            Transformer::EVENT_PRE_TRANSFORM
-        );
+        foreach ($project->getVersions() as $version) {
+            $apiSets = $version->getDocumentationSets()->filter(ApiSetDescriptor::class);
+            foreach ($apiSets as $apiSet) {
```

Comment:
Just to clarify, using multiple documentation sets would still not work because it is not fully supported when it comes to linking and even writing to the correct folder; but this will ensure the documentation set is propagated throughout the chain, which allowed me to remove findElement from the Project